src/App.tsx

```diff
@@ -17,18 +59,40 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                status={status}
+                query={query}
+                onQueryChange={setQuery}
+                onStatusChanged={setStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                isLoadTodos
+                  ? (
+                    <TodoList
+                      todos={filteredTodos}
+                      selectedTodoId={todoId}
+                      selectTodo={(todoID: number) => {
+                        setTodoId(todoID);
+                      }}
+                    />
+                  )
+                  : <Loader />
+              }
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {todoId !== 0 && (
+        <TodoModal
+          todo={[...todos].find(todo => todo.id === todoId)}
```

Comment:
Replace '[...todos].find(todo => todo.id === todoId)' with 'todos.find(todo => todo.id === todoId)' as there is no need to create a shallow copy of the todos array.